pynet/models/vae/vae.py

```diff
@@ -392,23 +395,28 @@ def __init__(self, input_channels, input_dim, conv_flts, dense_hidden_dims,
         # TODO: Not working well
         # self.kernel_initializer()
 
+    def _encode(self, x):
+        return self.encode(x)
+
+    def _decode(self, x):
+        return self.decode(x)
+
```

Comment:
Why adding this private methods?
You can define private networks self._encode & public method to access to this networks self.encode(self, x).